### ISSUE TITLE
`Layout/SpaceAroundOperators` should either always or never ignore assigments

### ISSUE THREAD

**Issue by deivid-rodriguez** [rubocop-hq/rubocop#7344]
## Expected behavior

I have code similar to this in my library:

```ruby
@env_shebang         = options[:env_shebang]
@gem_home            = options[:install_dir]
@only_install_dir    = options[:only_install_dir]
```

And I'm using the `Layout/SpaceAroundOperators` cop, which by default seems to allow spaces for alignment when the operator is an assignment.

Now, one contributor changes the code to this:

```ruby
@env_shebang         = options[:env_shebang]
if options[:install_as_default]
  @gem_home          = Gem.default_gems_dir
else
  @gem_home          = options[:install_dir]
end
@only_install_dir    = options[:only_install_dir]
```

My contributor is trying to keep the existing style, but rubocop complains:

```
Inspecting 1 file
C

Offenses:

test.rb:7:22: C: Layout/SpaceAroundOperators: Operator = should be surrounded by a single space.
  @gem_home          = Gem.default_gems_dir
                     ^
test.rb:9:22: C: Layout/SpaceAroundOperators: Operator = should be surrounded by a single space.
  @gem_home          = options[:install_dir]
                     ^

1 file inspected, 2 offenses detected
```

So I'm like "I'll just remove any spacing here", no spacing exceptions. So I enable:

```yaml
AllCops:
  DisabledByDefault: true

Layout/SpaceAroundOperators:
  AllowForAlignment: false
  Enabled: true
```

I would expect `rubocop` to signal all redundant space in the block, not only the above two offenses.

## Actual behavior

However, `rubocop` gives me the exact same result as before, it doesn't complain for any lines except for the previous two.

I'm aware that there's `Layout/ExtraSpacing` that would complain here, but I have intentionally disabled that one.

My point is that if "extra space around assigment symbol" is considered a job for the `Layout/ExtraSpacing` cop, `Layout/SpaceAroundOperators` should never complain on assigments.

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ [bundle exec] rubocop -V
0.74.0 (using Parser 2.6.4.0, running on ruby 2.6.4 x86_64-linux)
```

**Comment by deivid-rodriguez** [comment]
Hello! :wave: 

I'm always confused when the stale bot closes my issues. Is there any action that I'm supposed to take if I'm still interested in the issue?

**Comment by marcandre** [comment]
I'm with you... I think if you or anyone else reacts, the bot will reset the counter. 
I've reopened it for you.

**Comment by marcandre** [comment]
Have you considered:

```
@env_shebang         = options[:env_shebang]
@gem_home            = if options[:install_as_default]
                         Gem.default_gems_dir
                       else
                         options[:install_dir]
                       end
@only_install_dir    = options[:only_install_dir]

**Comment by deivid-rodriguez** [comment]
Thanks!

Actually, I just tried it again and it seems fixed now. I don't get any offenses at all from `Layout/SpaceAroundOperators`.

I should've verified this before complaining, my bad. I had been confused about the stale bot other times, so I decided to ask this time :)

Closing this!
